src/uu/seq/src/seq.rs

```diff
@@ -43,124 +44,57 @@ struct SeqOptions {
-                Ok(n) => {
-                    // If `s` is '-0', then `parse()` returns
-                    // `BigInt::zero()`, but we need to return
-                    // `Number::MinusZero` instead.
-                    if n == BigInt::zero() && is_neg {
-                        Ok(Number::MinusZero)
-                    } else {
-                        Ok(Number::BigInt(n))
-                    }
-                }
-                Err(_) => match s.parse::<f64>() {
-                    Ok(value) if value.is_nan() => Err(format!(
-                    "invalid 'not-a-number' argument: {}\nTry '{} --help' for more information.",
-                    s.quote(),
-                    uucore::execution_phrase(),
-                )),
-                    Ok(value) => Ok(Number::F64(value)),
-                    Err(_) => Err(format!(
-                    "invalid floating point argument: {}\nTry '{} --help' for more information.",
-                    s.quote(),
-                    uucore::execution_phrase(),
-                )),
-                },
-            },
-        }
-    }
-}
-
 /// A range of integers.
 ///
 /// The elements are (first, increment, last).
-type RangeInt = (BigInt, BigInt, BigInt);
+type RangeInt = (ExtendedBigInt, ExtendedBigInt, ExtendedBigInt);
 
-/// A range of f64.
+/// A range of floats.
 ///
 /// The elements are (first, increment, last).
-type RangeF64 = (f64, f64, f64);
+type RangeFloat = (ExtendedBigDecimal, ExtendedBigDecimal, ExtendedBigDecimal);
+
+/// Terminate the process with error code 1.
+///
+/// Before terminating the process, this function prints an error
+/// message that depends on `arg` and `e`.
+///
+/// Although the signature of this function states that it returns a
+/// [`PreciseNumber`], it never reaches the return statement. It is just
+/// there to make it easier to use this function when unwrapping the
+/// result of calling [`str::parse`] when attempting to parse a
+/// [`PreciseNumber`].
+///
+/// # Examples
+///
+/// ```rust,ignore
+/// let s = "1.2e-3";
+/// s.parse::<PreciseNumber>.unwrap_or_else(|e| exit_with_error(s, e))
+/// ```
+fn exit_with_error(arg: &str, e: ParseNumberError) -> PreciseNumber {
+    match e {
+        ParseNumberError::Float => crash!(
+            1,
+            "invalid floating point argument: {}\nTry '{} --help' for more information.",
+            arg.quote(),
+            uucore::execution_phrase()
+        ),
+        ParseNumberError::Nan => crash!(
+            1,
+            "invalid 'not-a-number' argument: {}\nTry '{} --help' for more information.",
+            arg.quote(),
+            uucore::execution_phrase()
+        ),
+        ParseNumberError::Hex => crash!(
+            1,
+            "invalid hexadecimal argument: {}\nTry '{} --help' for more information.",
+            arg.quote(),
+            uucore::execution_phrase()
+        ),
+    }
+    #[allow(unreachable_code)]
+    PreciseNumber::one()
```

Comment:
You can remove this return value and change the function's return type to `!` to indicate that it will never return.

Reply:
Ah, thank you!